backend/tests/unit/graphql/test_graphql_query.py

```diff
@@ -641,6 +641,107 @@ async def test_double_nested_query(db: InfrahubDatabase, default_branch: Branch,
+                "hierarchical": True,
+                "attributes": [{"name": "name", "optional": False, "kind": "Text"}],
+                "relationships": [{"name": "devices", "peer": "InfraDevice", "cardinality": "many", "optional": True}],
+            }
+        ],
+        "nodes": [
+            {
+                "name": "Device",
+                "namespace": "Infra",
+                "attributes": [{"name": "name", "kind": "Text", "optional": False}],
+                "relationships": [
+                    {"name": "location", "peer": "LocationGeneric", "optional": False, "cardinality": "one"}
+                ],
+            },
+            {
+                "name": "Site",
+                "namespace": "Location",
+                "inherit_from": ["LocationGeneric"],
+                "attributes": [{"name": "description", "optional": False, "kind": "Text"}],
+            },
+        ],
+    }
+    schema = SchemaRoot(**raw_schema)
+    schema_branch = registry.schema.register_schema(schema=schema, branch=default_branch.name)
+
+    site_schema = schema_branch.get_node(name="LocationSite")
+    device_schema = schema_branch.get_node(name="InfraDevice")
+
+    site1 = await Node.init(db=db, schema=site_schema, branch=default_branch)
+    await site1.new(db=db, name="site1", description="test")
+    await site1.save(db=db)
+
+    device1 = await Node.init(db=db, schema=device_schema, branch=default_branch)
+    await device1.new(db=db, name="device1", location=site1)
+    await device1.save(db=db)
+
+    device2 = await Node.init(db=db, schema=device_schema, branch=default_branch)
+    await device2.new(db=db, name="device2", location=site1)
+    await device2.save(db=db)
+
+    query = """
+    fragment LocationData on LocationSite {
+        name {
+            value
+        }
+        devices {
+            edges {
+                node {
+                    name {
+                        value
+                    }
+                }
+            }
+        }
+    }
+
+    query {
+        InfraDevice {
+            edges {
+                node {
+                    name {
+                        value
+                    }
+                    location {
+                        node {
+                            ... LocationData
+                        }
+                    }
+                }
+            }
+        }
+    }
+    """
+    gql_params = prepare_graphql_params(
+        db=db, include_mutation=False, include_subscription=False, branch=default_branch
+    )
+    result = await graphql(
+        schema=gql_params.schema,
+        source=query,
+        context_value=gql_params.context,
+        root_value=None,
+        variable_values={},
+    )
+
+    assert result.errors is None
+
+    result_per_name = {
+        result["node"]["name"]["value"]: result["node"] for result in result.data["InfraDevice"]["edges"]
+    }
+    assert sorted(result_per_name.keys()) == ["device1", "device2"]
```

Comment:
might be nice to check all the data in this response while we have it
verified this passes locally

```suggestion
    assert sorted(result_per_name.keys()) == ["device1", "device2"]
    expected_location_data = {
        "node": {
            "name": {"value": "site1"},
            "devices": {"edges": [{"node": {"name": {"value": "device1"}}}, {"node": {"name": {"value": "device2"}}}]},
        }
    }
    assert result.data["InfraDevice"]["edges"][0]["node"]["location"] == expected_location_data
    assert result.data["InfraDevice"]["edges"][1]["node"]["location"] == expected_location_data
```